test/index.js

```diff
@@ -273,31 +273,61 @@ describe('logger', () => {
         expect(this.output, 'to equal', fixtures[title]);
     });
 
-    it('should log an unhandled error', async function () {
-        const title = this.test.fullTitle();
-        const createLogger = this.createLogger(title);
+    describe('when an error is thrown', () => {
+        it('should log the error', async function () {
+            const title = this.test.fullTitle();
+            const createLogger = this.createLogger(title);
 
-        const logger = createLogger();
+            const logger = createLogger();
 
-        const context = {
-            method: 'GET',
-            originalUrl: '/'
-        };
+            const context = {
+                method: 'GET',
+                originalUrl: '/'
+            };
 
-        const next = () => {
-            const error = new Error();
-            error.stack = 'Error\n    at stack';
+            const next = () => {
+                const error = new Error();
+                error.stack = 'Error\n    at stack';
 
-            throw error;
-        };
+                throw error;
+            };
 
-        try {
-            await logger(context, next);
-        } catch (error) {
-            // Not interested in this one
-        }
+            try {
+                await logger(context, next);
+            } catch (error) {
+                // Not interested in this one
+            }
 
-        expect(this.output, 'to equal', fixtures[title]);
+            expect(this.output, 'to equal', fixtures[title]);
+        });
+
+        it('should log the error status if avaiable', async function () {
+            const title = this.test.fullTitle();
+            const createLogger = this.createLogger(title);
+
+            const logger = createLogger();
+
+            const context = {
+                method: 'GET',
+                originalUrl: '/'
+            };
+
+            const next = () => {
+                const error = new Error('GatewayTimeout');
```

Comment:
Looks like the formatting is different only because of overwriting `.stack` in the test cases. We need to make sure that it produces the same output for all environments, though.